docs/setup-and-configuration.md

```diff
@@ -11,5 +11,486 @@ A client is usually some system that you want to able to send request to the Ope
+    > **Note**: suggested roles will only appear as you type, if they already exist in the OpenHIM.
+- **Certificate** - The certificate field is used when the OpenHIM core is running using mutual TLS authentication and needs to authenticate requests coming from the client. 
+By default, the OpenHIM core uses mutual TLS authentication
+- **Basic Auth Password** - The password field is used when the OpenHIM core is running in basic auth mode and does not require a certificate, it does however require a password.
+
+### How to add clients
+> **Note**: All fields marked with a * or ** indicates a mandatory field.
+
+    * - Indicates a required field which means that it cannot be left blank
+    ** - Indicates that one of the fields are required
+
+1. Logon to your OpenHIM console
+1. Click on Clients found in the left navigation menu.
+1. Click on the button labelled “+ Client” to open a popup window where you will be able to supply the client details.
+1. Capture the client details
+1. Assign an existing role or enter a name for a new role which will be created and linked to this client
+1. You may choose to make use of a basic auth password or client certificate, depending on your OpenHIM configuration. If basic auth is enabled in the OpenHIM core configuration, then only a password is required and if mutual TLS authentication is enabled, then only a client certificate is required:
+
+    -  *Certificate*: You may choose one of the available client certificates from the Client Certificate drop down. 
+        > **Note**: This will only be possible if you have already configured one more more client certificates in the OpenHIM Console.
+    - *Basic Auth Password*: Enter a Basic Auth Password and confirm it by retyping it in the Confirm Password textbox
+1. Click on the Save Changes button to save your new client
+
+> **Note**: When a client certificate is added or updated in the Certificates component of the OpenHIM, the OpenHIM console will inform the OpenHIM administrator that a server restart is required. This is for the new certificate to be applied correctly. The user can either decide to manually restart the server at a later time or to click the red “Restart Server Now!” button.
+
+### How to remove clients
+
+1. Logon to your OpenHIM console
+1. Click on Clients found in the left navigation menu.
+1. Locate the client to be removed and click on the red X button on the far right
+1. You will be prompted to confirm your action to delete the chosen client
+1. Click on the Delete button
+
+> **Note**: This action automatically deletes the role if the role was created primarily for the client that is busy being deleted with no other clients sharing the same role.
+
+### How to edit clients
+
+1. Logon to your OpenHIM console
```

Comment:
Logon is not a term used on the console - please use either Sign in or Log in for all occurences